### PR TITLE
mintSign(0x657bb113) fix + add contract address screen to every method

### PR DESCRIPTION
**Pull request by Z4karia**

### CODE REVIEW

**Comment by clcmusic** [comment]
Note: hard-coding the contract address check was removed per the product requirements. Our use case is to deploy many smart contracts over time that are copies of one another. Each time that happens, the CAL is updated to include these new contract addresses, which acts as an "allowlist" of contract addresses that can be used to interact with the plugin.

Regarding [this previous comment](https://github.com/LedgerHQ/app-plugin-nft/pull/11#discussion_r1131078358) from a previous PR, one of the screens will now show the contract address that a user is interacting with, so this should satisfy the requirement of "clear signing" because the user can verify the exact smart contract they are interacting with.

**Comment by vforgeoux-ledger** [comment]
@Z4karia can you please bump the version to 1.6.0 ?

**Review by apaillier-ledger — APPROVED** [review]

File: src/handle_query_contract_ui.c [hunk 1]
```diff
@@ -20,7 +20,20 @@ static void set_payable_amount_ui(ethQueryContractUI_t *msg, context_t *context)
 static void set_amount_ui(ethQueryContractUI_t *msg, context_t *context) {
     strlcpy(msg->title, "Quantity", msg->titleLength);
 
-    amountToString(context->amount, sizeof(context->amount), 0, "", msg->msg, msg->msgLength);
+    if (context->selectorIndex == MINT_SIGN_V2) {
+        // Convert the amount to a uint32_t
+        uint32_t amountInt = U4BE(context->amount, PARAMETER_LENGTH - sizeof(amountInt));
+        // Multiply by the amount by the number of tokens (there should never be a transaction with
+        // multiple tokens with different amounts)
+        amountInt *= context->nb_tokens;
```
Comment (yhql) [inline src/handle_query_contract_ui.c:28]: Better check for overflows here, e.g. using `__builtin_umul_overflow` from https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html

File: src/handle_query_contract_ui.c [hunk 2]
```diff
@@ -20,7 +20,20 @@ static void set_payable_amount_ui(ethQueryContractUI_t *msg, context_t *context)
 static void set_amount_ui(ethQueryContractUI_t *msg, context_t *context) {
     strlcpy(msg->title, "Quantity", msg->titleLength);
 
-    amountToString(context->amount, sizeof(context->amount), 0, "", msg->msg, msg->msgLength);
+    if (context->selectorIndex == MINT_SIGN_V2) {
+        // Convert the amount to a uint32_t
+        uint32_t amountInt = U4BE(context->amount, PARAMETER_LENGTH - sizeof(amountInt));
```
Comment (jmartins-ledger) [inline src/handle_query_contract_ui.c:25]: When extracting a `uint16_t/uint32_t` value from a parameter these functions should be used https://github.com/LedgerHQ/ethereum-plugin-sdk/blob/b9777e7a81a9c33b3f8997ffa23396fa2b3f861d/include/eth_internals.c#L263-L272

**Pull request merged**